### ISSUE TITLE
Data Audit Gameplan

### ISSUE THREAD

**Issue by thermokarst** [codeforanchorage/trailstranscoder#4]
Seems like now would be a good time to start digging into the data --- what trails are missing various attributes, are there trails that should be scrubbed from the set, missing trail names, and so on. Maybe it makes sense to split up the audit by source, and track each audit in separate GH issues + corresponding git branches. We could open a PR for each branch, track the discussion in the issue, and make any changes to the `scripts/edits` directory in those branches/PRs. There will probably be a mix of comments in the issues like "all of the kincaid trails are missing surface info" (I don't know if that is the case, just a scenario), then we can decide how best to apply our edits to that particular situation. Any thoughts? Before we proceed with that, we should probably add in the other CSV that we discussed at our meeting last week, although I can't recall what we decided to name it. Thanks!

**Comment by brendanbabb** [comment]
I like the idea for global edits, like changing all the surface types at once for Kincaid. Something to think about further down the line, is how do we do individual edits? How can we give new people access to edit either through github or exposing the CSV some way and then even further for edits from people out on the trails.

**Comment by thermokarst** [comment]
I think for now it makes sense to rely on contributions via GitHub --- since these scripts aren't a platform. If this becomes something more like an interactive service down the line (or precipitates a service-like project, more specifically), then I think we should consider this in greater detail.

It seems easiest to rely on folks submitting PR's with updates/changes to the global edits files (e.g., the suppressions table: https://github.com/codeforanchorage/trailstranscoder/blob/master/scripts/edits/suppressions.csv).

**Comment by brendanbabb** [comment]
Sounds good. Cool on the CSV in place editing.

**Comment by mkbrk** [comment]
I agree with the general approach of dealing with data audits one source at a time. That seems like a logical way to break it up, given that someone might know a lot about one source and very little about another. I also like the idea of leveraging Github to keep us from stepping on each other. @thermokarst the method your outlined sounds like it would work for that.

Here are some additional thoughts:

Our goal should be to leave the upstream data sources _untouched_. Any edits we make have to be downstream of the sources. We should make the "fixed" sources available via GeoJSON, so that they can be folded back into the source files if desired. 

I believe we need to support three types of updates to data: 
- **suppressing** trail segments (i.e. the source says a trail is there, but there isn't)
- **updating properties** of trail segments (i.e. adding lighting attributes, trails direction, fixing typos, etc.)
- **adding trails and segments**

Now, the hardest part of all this is _deciding on a unique identifier_ for trails and segments. This is actually pretty difficult. I propose a two-part approach to this:
1. Trails tend to be named. I propose a three-part key made up of source (e.g. "Muni"), system (e.g. "Kincaid Ski Trails"), and trail name (e.g. "Mizes Loop"). Note that a single source/system/name triad _probably_ refers to more than one trail segment in our dataset. 
2. Trail segments that aren't part of a named trail can be identified by a _hash_ of the geometry of the segment. To calculate the hash, we just need to establish a canonical way to represent the geometry (e.g. remove all spaces from the JSON array) and then run MD5 over it. 

Now, update scenarios:
1. **Suppressing** trail segments. I propose we keep a simple list of MD5 hashes that need to be _excluded_ from the final data set. We could keep one list per source, as per the initial suggestion.
2. **Updating** properties of trails. We should be able to use the source/system/name triad for these. If so, we can use a simple CSV to splice in updated properties (again, one per source):

```
#the idea here is that the source, system, and name fields are used to match
#segments in the source files and the remaining fields are used to update the data
source,system,name,lighting,skitype,surface,...
"Muni","Kincaid Ski Trails","Mizes Loop","No",,"Earth", ...
```
1. **Adding** trails and segments. I think GeoJSON is the way to go, simply because of the difficulty of cramming geometry into CSV. Handheld GPSs tend to emit GPX files (at least mine does), which we could easily convert to GeoJSON and then annotate (by hand, probably) with known properties (e.g. surface, etc.).

This is a pretty big topic. Let's definitely put it on the agenda for Tuesday.
